src/buffer_writer.c

```diff
@@ -57,19 +59,65 @@ bool buffer_writer_write_value(buffer_writer_t *buffer_writer, const void *value
     return true;
 }
 
-bool buffer_writer_write_uint8(buffer_writer_t *buffer_writer, uint8_t value)
+bool MUSTCHECK buffer_writer_write_buffer(buffer_writer_t *buffer_writer, const buffer_writer_t *srcbuf)
+{
+    if(NULL == srcbuf || NULL == srcbuf->buffer)
+        return false;
+    return buffer_writer_write_value(buffer_writer, srcbuf->buffer->data, srcbuf->write_offset);
```

Comment:
This function seems unused, but should it really pass the `write_offset` as length to `buffer_writer_write_value`?